{
	"documenttitle": {
		"text": "Red Hat Security Advisory: rh-python36-python security, bug fix, and enhancement update",
		"lang": "en"
	},
	"DocumentType": "Security Advisory",
	"documentpublisher": {
		"type": "Vendor",
		"ContactDetails": "secalert@redhat.com",
		"IssuingAuthority": "Red Hat Product Security"
	},
	"documenttracking": {
		"identification": {
			"ID": "RHSA-2019:3725"
		},
		"Status": "Final",
		"Version": "1",
		"revisionhistory": {
			"revision": {
				"Number": "1",
				"Date": "2021-02-02T04:20:00Z",
				"Description": "Current version"
			}
		},
		"InitialReleaseDate": "2019-11-06T09:47:00Z",
		"CurrentReleaseDate": "2021-02-02T04:20:00Z",
		"generator": {
			"Engine": "Red Hat SDEngine 3.0.0",
			"Date": "2021-08-23T04:47:00Z"
		}
	},
	"documentnotes": {
		"note": [
			{
				"text": "\nAn update for rh-python36-python is now available for Red Hat Software Collections.\n\nRed Hat Product Security has rated this update as having a security impact of Moderate. A Common Vulnerability Scoring System (CVSS) base score, which gives a detailed severity rating, is available for each vulnerability from the CVE link(s) in the References section.\n    ",
				"title": "Topic",
				"type": "Summary",
				"ordinal": "1",
				"lang": "en"
			},
			{
				"text": "\nPython is an interpreted, interactive, object-oriented programming language, which includes modules, classes, exceptions, very high level dynamic data types and dynamic typing. Python supports interfaces to many system calls and libraries, as well as to various windowing systems.\n\nThe following packages have been upgraded to a later upstream version: rh-python36-python (3.6.9). (BZ#1709344)\n\nSecurity Fix(es):\n\n* python: DOS via regular expression backtracking in difflib.IS_LINE_JUNK method in difflib (CVE-2018-1061)\n\n* python: Missing salt initialization in _elementtree.c module (CVE-2018-14647)\n\n* python: Cookie domain check returns incorrect results (CVE-2018-20852)\n\n* python: NULL pointer dereference using a specially crafted X509 certificate (CVE-2019-5010)\n\n* python: CRLF injection via the query part of the url passed to urlopen() (CVE-2019-9740)\n\n* python: CRLF injection via the path part of the url passed to urlopen() (CVE-2019-9947)\n\n* python: email.utils.parseaddr wrongly parses email addresses (CVE-2019-16056)\n\n* python: DOS via regular expression catastrophic backtracking in apop() method in pop3lib (CVE-2018-1060)\n\n* python: Integer overflow in Modules/_pickle.c allows for memory exhaustion if serializing gigabytes of data (CVE-2018-20406)\n\nFor more details about the security issue(s), including the impact, a CVSS score, acknowledgments, and other related information, refer to the CVE page(s) listed in the References section.\n\nBug Fix(es):\n\n* urlsplit doesn't accept a NFKD hostname with a port number (BZ#1709340)\n\n* rh-python36 cannot unpickle datetime.date objects (BZ#1749103)\n    ",
				"title": "Details",
				"type": "General",
				"ordinal": "2",
				"lang": "en"
			},
			{
				"text": "This content is licensed under the Creative Commons Attribution 4.0 International License (https://creativecommons.org/licenses/by/4.0/). If you distribute this content, or a modified version of it, you must provide attribution to Red Hat Inc. and provide a link to the original.",
				"title": "Terms of Use",
				"type": "Legal Disclaimer",
				"ordinal": "3",
				"lang": "en"
			}
		]
	},
	"documentdistribution": {
		"text": "Copyright Â© 2021 Red Hat, Inc. All rights reserved.",
		"lang": "en"
	},
	"aggregateseverity": {
		"text": "Moderate",
		"namespace": "https://access.redhat.com/security/updates/classification/"
	},
	"documentreferences": {
		"reference": [
			{
				"type": "Self",
				"URL": "https://access.redhat.com/errata/RHSA-2019:3725",
				"Description": "https://access.redhat.com/errata/RHSA-2019:3725"
			},
			{
				"type": "External",
				"URL": "https://access.redhat.com/security/updates/classification/#moderate",
				"Description": "https://access.redhat.com/security/updates/classification/#moderate"
			}
		]
	},
	"vulnerability": [
		{
			"ordinal": "1",
			"xmlns": "http://www.icasi.org/CVRF/schema/vuln/1.1",
			"notes": {
				"note": {
					"text": "python: DOS via regular expression catastrophic backtracking in apop() method in pop3lib",
					"title": "Vulnerability Description",
					"type": "General",
					"ordinal": "1",
					"lang": "en"
				}
			},
			"ReleaseDate": "2018-03-14T00:00:00Z",
			"involvements": {
				"involvement": {
					"party": "Vendor",
					"status": "Completed"
				}
			},
			"CVE": "CVE-2018-1060",
			"remediations": {
				"remediation": {
					"type": "Vendor Fix",
					"description": {
						"text": "\nFor details on how to apply this update, which includes the changes described in this advisory, refer to:\n\nhttps://access.redhat.com/articles/11258\n    ",
						"lang": "en"
					},
					"URL": "https://access.redhat.com/errata/RHSA-2019:3725"
				}
			},
			"references": {
				"reference": [
					{
						"URL": "https://docs.python.org/3.6/whatsnew/changelog.html#python-3-6-5-final",
						"Description": "https://docs.python.org/3.6/whatsnew/changelog.html#python-3-6-5-final"
					},
					{
						"URL": "https://access.redhat.com/security/cve/CVE-2018-1060",
						"Description": "CVE-2018-1060"
					},
					{
						"URL": "https://bugzilla.redhat.com/show_bug.cgi?id=1549191",
						"Description": "bz#1549191: CVE-2018-1060 python: DOS via regular expression catastrophic backtracking in apop() method in pop3lib"
					}
				]
			},
			"threats": {
				"threat": {
					"type": "Impact",
					"Description": "Low"
				}
			},
			"DiscoveryDate": "2018-02-26T00:00:00Z",
			"productstatuses": {
				"status": {
					"type": "Fixed",
					"ProductID": [
						"6Server-RHSCL-3.3:rh-python36-python-0:3.6.9-2.el6",
						"6Workstation-RHSCL-3.3:rh-python36-python-0:3.6.9-2.el6",
						"7Server-Alt-RHSCL-3.3:rh-python36-python-0:3.6.9-2.el7",
						"7Server-RHSCL-3.3-7.5.Z:rh-python36-python-0:3.6.9-2.el7",
						"7Server-RHSCL-3.3-7.6.Z:rh-python36-python-0:3.6.9-2.el7",
						"7Server-RHSCL-3.3-7.7.Z:rh-python36-python-0:3.6.9-2.el7",
						"7Server-RHSCL-3.3:rh-python36-python-0:3.6.9-2.el7",
						"7Workstation-RHSCL-3.3:rh-python36-python-0:3.6.9-2.el7"
					]
				}
			},
			"acknowledgments": {
				"acknowledgment": {
					"Description": "Red Hat would like to thank the Python security response team for reporting this issue."
				}
			},
			"cvssscoresets": {
				"scoreset": {
					"BaseScore": "",
					"Vector": ""
				}
			}
		},
		{
			"ordinal": "2",
			"xmlns": "http://www.icasi.org/CVRF/schema/vuln/1.1",
			"notes": {
				"note": {
					"text": "python: DOS via regular expression backtracking in difflib.IS_LINE_JUNK method in difflib",
					"title": "Vulnerability Description",
					"type": "General",
					"ordinal": "1",
					"lang": "en"
				}
			},
			"ReleaseDate": "2018-03-14T00:00:00Z",
			"involvements": {
				"involvement": {
					"party": "Vendor",
					"status": "Completed"
				}
			},
			"CVE": "CVE-2018-1061",
			"remediations": {
				"remediation": {
					"type": "Vendor Fix",
					"description": {
						"text": "\nFor details on how to apply this update, which includes the changes described in this advisory, refer to:\n\nhttps://access.redhat.com/articles/11258\n    ",
						"lang": "en"
					},
					"URL": "https://access.redhat.com/errata/RHSA-2019:3725"
				}
			},
			"references": {
				"reference": [
					{
						"URL": "https://docs.python.org/3.6/whatsnew/changelog.html#python-3-6-5-final",
						"Description": "https://docs.python.org/3.6/whatsnew/changelog.html#python-3-6-5-final"
					},
					{
						"URL": "https://access.redhat.com/security/cve/CVE-2018-1061",
						"Description": "CVE-2018-1061"
					},
					{
						"URL": "https://bugzilla.redhat.com/show_bug.cgi?id=1549192",
						"Description": "bz#1549192: CVE-2018-1061 python: DOS via regular expression backtracking in difflib.IS_LINE_JUNK method in difflib"
					}
				]
			},
			"threats": {
				"threat": {
					"type": "Impact",
					"Description": "Moderate"
				}
			},
			"DiscoveryDate": "2018-02-26T00:00:00Z",
			"productstatuses": {
				"status": {
					"type": "Fixed",
					"ProductID": [
						"6Server-RHSCL-3.3:rh-python36-python-0:3.6.9-2.el6",
						"6Workstation-RHSCL-3.3:rh-python36-python-0:3.6.9-2.el6",
						"7Server-Alt-RHSCL-3.3:rh-python36-python-0:3.6.9-2.el7",
						"7Server-RHSCL-3.3-7.5.Z:rh-python36-python-0:3.6.9-2.el7",
						"7Server-RHSCL-3.3-7.6.Z:rh-python36-python-0:3.6.9-2.el7",
						"7Server-RHSCL-3.3-7.7.Z:rh-python36-python-0:3.6.9-2.el7",
						"7Server-RHSCL-3.3:rh-python36-python-0:3.6.9-2.el7",
						"7Workstation-RHSCL-3.3:rh-python36-python-0:3.6.9-2.el7"
					]
				}
			},
			"acknowledgments": {
				"acknowledgment": {
					"Description": "Red Hat would like to thank the Python security response team for reporting this issue."
				}
			},
			"cvssscoresets": {
				"scoreset": {
					"BaseScore": "",
					"Vector": ""
				}
			}
		},
		{
			"ordinal": "3",
			"xmlns": "http://www.icasi.org/CVRF/schema/vuln/1.1",
			"notes": {
				"note": {
					"text": "python: Missing salt initialization in _elementtree.c module",
					"title": "Vulnerability Description",
					"type": "General",
					"ordinal": "1",
					"lang": "en"
				}
			},
			"ReleaseDate": "2018-09-22T00:00:00Z",
			"involvements": {
				"involvement": {
					"party": "Vendor",
					"status": "Completed"
				}
			},
			"CVE": "CVE-2018-14647",
			"remediations": {
				"remediation": {
					"type": "Vendor Fix",
					"description": {
						"text": "\nFor details on how to apply this update, which includes the changes described in this advisory, refer to:\n\nhttps://access.redhat.com/articles/11258\n    ",
						"lang": "en"
					},
					"URL": "https://access.redhat.com/errata/RHSA-2019:3725"
				}
			},
			"references": {
				"reference": [
					{
						"URL": "https://bugs.python.org/issue34623",
						"Description": "https://bugs.python.org/issue34623"
					},
					{
						"URL": "https://access.redhat.com/security/cve/CVE-2018-14647",
						"Description": "CVE-2018-14647"
					},
					{
						"URL": "https://bugzilla.redhat.com/show_bug.cgi?id=1631822",
						"Description": "bz#1631822: CVE-2018-14647 python: Missing salt initialization in _elementtree.c module"
					}
				]
			},
			"threats": {
				"threat": {
					"type": "Impact",
					"Description": "Moderate"
				}
			},
			"DiscoveryDate": "2018-09-21T00:00:00Z",
			"productstatuses": {
				"status": {
					"type": "Fixed",
					"ProductID": [
						"6Server-RHSCL-3.3:rh-python36-python-0:3.6.9-2.el6",
						"6Workstation-RHSCL-3.3:rh-python36-python-0:3.6.9-2.el6",
						"7Server-Alt-RHSCL-3.3:rh-python36-python-0:3.6.9-2.el7",
						"7Server-RHSCL-3.3-7.5.Z:rh-python36-python-0:3.6.9-2.el7",
						"7Server-RHSCL-3.3-7.6.Z:rh-python36-python-0:3.6.9-2.el7",
						"7Server-RHSCL-3.3-7.7.Z:rh-python36-python-0:3.6.9-2.el7",
						"7Server-RHSCL-3.3:rh-python36-python-0:3.6.9-2.el7",
						"7Workstation-RHSCL-3.3:rh-python36-python-0:3.6.9-2.el7"
					]
				}
			},
			"acknowledgments": {
				"acknowledgment": {
					"Description": "Red Hat would like to thank the Python Security Response Team for reporting this issue."
				}
			},
			"cvssscoresets": {
				"scoreset": {
					"BaseScore": "",
					"Vector": ""
				}
			}
		},
		{
			"ordinal": "4",
			"xmlns": "http://www.icasi.org/CVRF/schema/vuln/1.1",
			"notes": {
				"note": {
					"text": "python: Integer overflow in Modules/_pickle.c allows for memory exhaustion if serializing gigabytes of data",
					"title": "Vulnerability Description",
					"type": "General",
					"ordinal": "1",
					"lang": "en"
				}
			},
			"ReleaseDate": "2018-09-21T00:00:00Z",
			"involvements": {
				"involvement": {
					"party": "Vendor",
					"status": "Completed"
				}
			},
			"CVE": "CVE-2018-20406",
			"remediations": {
				"remediation": {
					"type": "Vendor Fix",
					"description": {
						"text": "\nFor details on how to apply this update, which includes the changes described in this advisory, refer to:\n\nhttps://access.redhat.com/articles/11258\n    ",
						"lang": "en"
					},
					"URL": "https://access.redhat.com/errata/RHSA-2019:3725"
				}
			},
			"references": {
				"reference": [
					{
						"URL": "https://python-security.readthedocs.io/vuln/pickle-load-dos.html",
						"Description": "https://python-security.readthedocs.io/vuln/pickle-load-dos.html"
					},
					{
						"URL": "https://access.redhat.com/security/cve/CVE-2018-20406",
						"Description": "CVE-2018-20406"
					},
					{
						"URL": "https://bugzilla.redhat.com/show_bug.cgi?id=1664509",
						"Description": "bz#1664509: CVE-2018-20406 python: Integer overflow in Modules/_pickle.c allows for memory exhaustion if serializing gigabytes of data"
					}
				]
			},
			"threats": {
				"threat": {
					"type": "Impact",
					"Description": "Low"
				}
			},
			"DiscoveryDate": "2019-01-04T00:00:00Z",
			"productstatuses": {
				"status": {
					"type": "Fixed",
					"ProductID": [
						"6Server-RHSCL-3.3:rh-python36-python-0:3.6.9-2.el6",
						"6Workstation-RHSCL-3.3:rh-python36-python-0:3.6.9-2.el6",
						"7Server-Alt-RHSCL-3.3:rh-python36-python-0:3.6.9-2.el7",
						"7Server-RHSCL-3.3-7.5.Z:rh-python36-python-0:3.6.9-2.el7",
						"7Server-RHSCL-3.3-7.6.Z:rh-python36-python-0:3.6.9-2.el7",
						"7Server-RHSCL-3.3-7.7.Z:rh-python36-python-0:3.6.9-2.el7",
						"7Server-RHSCL-3.3:rh-python36-python-0:3.6.9-2.el7",
						"7Workstation-RHSCL-3.3:rh-python36-python-0:3.6.9-2.el7"
					]
				}
			},
			"acknowledgments": {
				"acknowledgment": {
					"Description": ""
				}
			},
			"cvssscoresets": {
				"scoreset": {
					"BaseScore": "",
					"Vector": ""
				}
			}
		},
		{
			"ordinal": "5",
			"xmlns": "http://www.icasi.org/CVRF/schema/vuln/1.1",
			"notes": {
				"note": {
					"text": "python: Cookie domain check returns incorrect results",
					"title": "Vulnerability Description",
					"type": "General",
					"ordinal": "1",
					"lang": "en"
				}
			},
			"ReleaseDate": "2018-10-31T00:00:00Z",
			"involvements": {
				"involvement": {
					"party": "Vendor",
					"status": "Completed"
				}
			},
			"CVE": "CVE-2018-20852",
			"remediations": {
				"remediation": {
					"type": "Vendor Fix",
					"description": {
						"text": "\nFor details on how to apply this update, which includes the changes described in this advisory, refer to:\n\nhttps://access.redhat.com/articles/11258\n    ",
						"lang": "en"
					},
					"URL": "https://access.redhat.com/errata/RHSA-2019:3725"
				}
			},
			"references": {
				"reference": [
					{
						"URL": "https://access.redhat.com/security/cve/CVE-2018-20852",
						"Description": "CVE-2018-20852"
					},
					{
						"URL": "https://bugzilla.redhat.com/show_bug.cgi?id=1740347",
						"Description": "bz#1740347: CVE-2018-20852 python: Cookie domain check returns incorrect results"
					}
				]
			},
			"threats": {
				"threat": {
					"type": "Impact",
					"Description": "Moderate"
				}
			},
			"DiscoveryDate": "2019-07-14T00:00:00Z",
			"productstatuses": {
				"status": {
					"type": "Fixed",
					"ProductID": [
						"6Server-RHSCL-3.3:rh-python36-python-0:3.6.9-2.el6",
						"6Workstation-RHSCL-3.3:rh-python36-python-0:3.6.9-2.el6",
						"7Server-Alt-RHSCL-3.3:rh-python36-python-0:3.6.9-2.el7",
						"7Server-RHSCL-3.3-7.5.Z:rh-python36-python-0:3.6.9-2.el7",
						"7Server-RHSCL-3.3-7.6.Z:rh-python36-python-0:3.6.9-2.el7",
						"7Server-RHSCL-3.3-7.7.Z:rh-python36-python-0:3.6.9-2.el7",
						"7Server-RHSCL-3.3:rh-python36-python-0:3.6.9-2.el7",
						"7Workstation-RHSCL-3.3:rh-python36-python-0:3.6.9-2.el7"
					]
				}
			},
			"acknowledgments": {
				"acknowledgment": {
					"Description": ""
				}
			},
			"cvssscoresets": {
				"scoreset": {
					"BaseScore": "",
					"Vector": ""
				}
			}
		},
		{
			"ordinal": "6",
			"xmlns": "http://www.icasi.org/CVRF/schema/vuln/1.1",
			"notes": {
				"note": {
					"text": "python: NULL pointer dereference using a specially crafted X509 certificate",
					"title": "Vulnerability Description",
					"type": "General",
					"ordinal": "1",
					"lang": "en"
				}
			},
			"ReleaseDate": "2019-01-15T00:00:00Z",
			"involvements": {
				"involvement": {
					"party": "Vendor",
					"status": "Completed"
				}
			},
			"CVE": "CVE-2019-5010",
			"remediations": {
				"remediation": {
					"type": "Vendor Fix",
					"description": {
						"text": "\nFor details on how to apply this update, which includes the changes described in this advisory, refer to:\n\nhttps://access.redhat.com/articles/11258\n    ",
						"lang": "en"
					},
					"URL": "https://access.redhat.com/errata/RHSA-2019:3725"
				}
			},
			"references": {
				"reference": [
					{
						"URL": "https://python-security.readthedocs.io/vuln/ssl-crl-dps-dos.html",
						"Description": "https://python-security.readthedocs.io/vuln/ssl-crl-dps-dos.html"
					},
					{
						"URL": "https://access.redhat.com/security/cve/CVE-2019-5010",
						"Description": "CVE-2019-5010"
					},
					{
						"URL": "https://bugzilla.redhat.com/show_bug.cgi?id=1666519",
						"Description": "bz#1666519: CVE-2019-5010 python: NULL pointer dereference using a specially crafted X509 certificate"
					}
				]
			},
			"threats": {
				"threat": {
					"type": "Impact",
					"Description": "Moderate"
				}
			},
			"DiscoveryDate": "2019-01-15T00:00:00Z",
			"productstatuses": {
				"status": {
					"type": "Fixed",
					"ProductID": [
						"6Server-RHSCL-3.3:rh-python36-python-0:3.6.9-2.el6",
						"6Workstation-RHSCL-3.3:rh-python36-python-0:3.6.9-2.el6",
						"7Server-Alt-RHSCL-3.3:rh-python36-python-0:3.6.9-2.el7",
						"7Server-RHSCL-3.3-7.5.Z:rh-python36-python-0:3.6.9-2.el7",
						"7Server-RHSCL-3.3-7.6.Z:rh-python36-python-0:3.6.9-2.el7",
						"7Server-RHSCL-3.3-7.7.Z:rh-python36-python-0:3.6.9-2.el7",
						"7Server-RHSCL-3.3:rh-python36-python-0:3.6.9-2.el7",
						"7Workstation-RHSCL-3.3:rh-python36-python-0:3.6.9-2.el7"
					]
				}
			},
			"acknowledgments": {
				"acknowledgment": {
					"Description": ""
				}
			},
			"cvssscoresets": {
				"scoreset": {
					"BaseScore": "",
					"Vector": ""
				}
			}
		},
		{
			"ordinal": "7",
			"xmlns": "http://www.icasi.org/CVRF/schema/vuln/1.1",
			"notes": {
				"note": {
					"text": "python: CRLF injection via the query part of the url passed to urlopen()",
					"title": "Vulnerability Description",
					"type": "General",
					"ordinal": "1",
					"lang": "en"
				}
			},
			"ReleaseDate": "2019-03-13T00:00:00Z",
			"involvements": {
				"involvement": {
					"party": "Vendor",
					"status": "Completed"
				}
			},
			"CVE": "CVE-2019-9740",
			"remediations": {
				"remediation": {
					"type": "Vendor Fix",
					"description": {
						"text": "\nFor details on how to apply this update, which includes the changes described in this advisory, refer to:\n\nhttps://access.redhat.com/articles/11258\n    ",
						"lang": "en"
					},
					"URL": "https://access.redhat.com/errata/RHSA-2019:3725"
				}
			},
			"references": {
				"reference": [
					{
						"URL": "https://access.redhat.com/security/cve/CVE-2019-9740",
						"Description": "CVE-2019-9740"
					},
					{
						"URL": "https://bugzilla.redhat.com/show_bug.cgi?id=1688169",
						"Description": "bz#1688169: CVE-2019-9740 python: CRLF injection via the query part of the url passed to urlopen()"
					}
				]
			},
			"threats": {
				"threat": {
					"type": "Impact",
					"Description": "Moderate"
				}
			},
			"DiscoveryDate": "2019-03-13T00:00:00Z",
			"productstatuses": {
				"status": {
					"type": "Fixed",
					"ProductID": [
						"6Server-RHSCL-3.3:rh-python36-python-0:3.6.9-2.el6",
						"6Workstation-RHSCL-3.3:rh-python36-python-0:3.6.9-2.el6",
						"7Server-Alt-RHSCL-3.3:rh-python36-python-0:3.6.9-2.el7",
						"7Server-RHSCL-3.3-7.5.Z:rh-python36-python-0:3.6.9-2.el7",
						"7Server-RHSCL-3.3-7.6.Z:rh-python36-python-0:3.6.9-2.el7",
						"7Server-RHSCL-3.3-7.7.Z:rh-python36-python-0:3.6.9-2.el7",
						"7Server-RHSCL-3.3:rh-python36-python-0:3.6.9-2.el7",
						"7Workstation-RHSCL-3.3:rh-python36-python-0:3.6.9-2.el7"
					]
				}
			},
			"acknowledgments": {
				"acknowledgment": {
					"Description": ""
				}
			},
			"cvssscoresets": {
				"scoreset": {
					"BaseScore": "",
					"Vector": ""
				}
			}
		},
		{
			"ordinal": "8",
			"xmlns": "http://www.icasi.org/CVRF/schema/vuln/1.1",
			"notes": {
				"note": {
					"text": "python: CRLF injection via the path part of the url passed to urlopen()",
					"title": "Vulnerability Description",
					"type": "General",
					"ordinal": "1",
					"lang": "en"
				}
			},
			"ReleaseDate": "2019-03-23T00:00:00Z",
			"involvements": {
				"involvement": {
					"party": "Vendor",
					"status": "Completed"
				}
			},
			"CVE": "CVE-2019-9947",
			"remediations": {
				"remediation": {
					"type": "Vendor Fix",
					"description": {
						"text": "\nFor details on how to apply this update, which includes the changes described in this advisory, refer to:\n\nhttps://access.redhat.com/articles/11258\n    ",
						"lang": "en"
					},
					"URL": "https://access.redhat.com/errata/RHSA-2019:3725"
				}
			},
			"references": {
				"reference": [
					{
						"URL": "https://access.redhat.com/security/cve/CVE-2019-9947",
						"Description": "CVE-2019-9947"
					},
					{
						"URL": "https://bugzilla.redhat.com/show_bug.cgi?id=1695572",
						"Description": "bz#1695572: CVE-2019-9947 python: CRLF injection via the path part of the url passed to urlopen()"
					}
				]
			},
			"threats": {
				"threat": {
					"type": "Impact",
					"Description": "Moderate"
				}
			},
			"DiscoveryDate": "2019-03-28T00:00:00Z",
			"productstatuses": {
				"status": {
					"type": "Fixed",
					"ProductID": [
						"6Server-RHSCL-3.3:rh-python36-python-0:3.6.9-2.el6",
						"6Workstation-RHSCL-3.3:rh-python36-python-0:3.6.9-2.el6",
						"7Server-Alt-RHSCL-3.3:rh-python36-python-0:3.6.9-2.el7",
						"7Server-RHSCL-3.3-7.5.Z:rh-python36-python-0:3.6.9-2.el7",
						"7Server-RHSCL-3.3-7.6.Z:rh-python36-python-0:3.6.9-2.el7",
						"7Server-RHSCL-3.3-7.7.Z:rh-python36-python-0:3.6.9-2.el7",
						"7Server-RHSCL-3.3:rh-python36-python-0:3.6.9-2.el7",
						"7Workstation-RHSCL-3.3:rh-python36-python-0:3.6.9-2.el7"
					]
				}
			},
			"acknowledgments": {
				"acknowledgment": {
					"Description": ""
				}
			},
			"cvssscoresets": {
				"scoreset": {
					"BaseScore": "",
					"Vector": ""
				}
			}
		},
		{
			"ordinal": "9",
			"xmlns": "http://www.icasi.org/CVRF/schema/vuln/1.1",
			"notes": {
				"note": {
					"text": "python: Undocumented local_file protocol allows remote attackers to bypass protection mechanisms",
					"title": "Vulnerability Description",
					"type": "General",
					"ordinal": "1",
					"lang": "en"
				}
			},
			"ReleaseDate": "2019-03-23T00:00:00Z",
			"involvements": {
				"involvement": {
					"party": "Vendor",
					"status": "Completed"
				}
			},
			"CVE": "CVE-2019-9948",
			"remediations": {
				"remediation": {
					"type": "Vendor Fix",
					"description": {
						"text": "\nFor details on how to apply this update, which includes the changes described in this advisory, refer to:\n\nhttps://access.redhat.com/articles/11258\n    ",
						"lang": "en"
					},
					"URL": "https://access.redhat.com/errata/RHSA-2019:3725"
				}
			},
			"references": {
				"reference": [
					{
						"URL": "https://access.redhat.com/security/cve/CVE-2019-9948",
						"Description": "CVE-2019-9948"
					},
					{
						"URL": "https://bugzilla.redhat.com/show_bug.cgi?id=1695570",
						"Description": "bz#1695570: CVE-2019-9948 python: Undocumented local_file protocol allows remote attackers to bypass protection mechanisms"
					}
				]
			},
			"threats": {
				"threat": {
					"type": "Impact",
					"Description": "Moderate"
				}
			},
			"DiscoveryDate": "2019-03-28T00:00:00Z",
			"productstatuses": {
				"status": {
					"type": "Fixed",
					"ProductID": [
						"6Server-RHSCL-3.3:rh-python36-python-0:3.6.9-2.el6",
						"6Workstation-RHSCL-3.3:rh-python36-python-0:3.6.9-2.el6",
						"7Server-Alt-RHSCL-3.3:rh-python36-python-0:3.6.9-2.el7",
						"7Server-RHSCL-3.3-7.5.Z:rh-python36-python-0:3.6.9-2.el7",
						"7Server-RHSCL-3.3-7.6.Z:rh-python36-python-0:3.6.9-2.el7",
						"7Server-RHSCL-3.3-7.7.Z:rh-python36-python-0:3.6.9-2.el7",
						"7Server-RHSCL-3.3:rh-python36-python-0:3.6.9-2.el7",
						"7Workstation-RHSCL-3.3:rh-python36-python-0:3.6.9-2.el7"
					]
				}
			},
			"acknowledgments": {
				"acknowledgment": {
					"Description": ""
				}
			},
			"cvssscoresets": {
				"scoreset": {
					"BaseScore": "",
					"Vector": ""
				}
			}
		},
		{
			"ordinal": "10",
			"xmlns": "http://www.icasi.org/CVRF/schema/vuln/1.1",
			"notes": {
				"note": {
					"text": "python: email.utils.parseaddr wrongly parses email addresses",
					"title": "Vulnerability Description",
					"type": "General",
					"ordinal": "1",
					"lang": "en"
				}
			},
			"ReleaseDate": "2018-07-19T00:00:00Z",
			"involvements": {
				"involvement": {
					"party": "Vendor",
					"status": "Completed"
				}
			},
			"CVE": "CVE-2019-16056",
			"remediations": {
				"remediation": {
					"type": "Vendor Fix",
					"description": {
						"text": "\nFor details on how to apply this update, which includes the changes described in this advisory, refer to:\n\nhttps://access.redhat.com/articles/11258\n    ",
						"lang": "en"
					},
					"URL": "https://access.redhat.com/errata/RHSA-2019:3725"
				}
			},
			"references": {
				"reference": [
					{
						"URL": "https://access.redhat.com/security/cve/CVE-2019-16056",
						"Description": "CVE-2019-16056"
					},
					{
						"URL": "https://bugzilla.redhat.com/show_bug.cgi?id=1749839",
						"Description": "bz#1749839: CVE-2019-16056 python: email.utils.parseaddr wrongly parses email addresses"
					}
				]
			},
			"threats": {
				"threat": {
					"type": "Impact",
					"Description": "Moderate"
				}
			},
			"DiscoveryDate": "2019-09-06T00:00:00Z",
			"productstatuses": {
				"status": {
					"type": "Fixed",
					"ProductID": [
						"6Server-RHSCL-3.3:rh-python36-python-0:3.6.9-2.el6",
						"6Workstation-RHSCL-3.3:rh-python36-python-0:3.6.9-2.el6",
						"7Server-Alt-RHSCL-3.3:rh-python36-python-0:3.6.9-2.el7",
						"7Server-RHSCL-3.3-7.5.Z:rh-python36-python-0:3.6.9-2.el7",
						"7Server-RHSCL-3.3-7.6.Z:rh-python36-python-0:3.6.9-2.el7",
						"7Server-RHSCL-3.3-7.7.Z:rh-python36-python-0:3.6.9-2.el7",
						"7Server-RHSCL-3.3:rh-python36-python-0:3.6.9-2.el7",
						"7Workstation-RHSCL-3.3:rh-python36-python-0:3.6.9-2.el7"
					]
				}
			},
			"acknowledgments": {
				"acknowledgment": {
					"Description": ""
				}
			},
			"cvssscoresets": {
				"scoreset": {
					"BaseScore": "",
					"Vector": ""
				}
			}
		}
	],
	"producttree": {
		"xmlns": "http://www.icasi.org/CVRF/schema/prod/1.1",
		"branch": [
			{
				"type": "Product Family",
				"name": "Red Hat Software Collections",
				"branch": [
					{
						"type": "Product Name",
						"name": "Red Hat Software Collections for Red Hat Enterprise Linux Server (v. 6)",
						"fullproductname": {
							"text": "Red Hat Software Collections for Red Hat Enterprise Linux Server (v. 6)",
							"productid": "6Server-RHSCL-3.3",
							"cpe": "cpe:/a:redhat:rhel_software_collections:3::el6"
						}
					},
					{
						"type": "Product Name",
						"name": "Red Hat Software Collections for Red Hat Enterprise Linux Workstation (v. 6)",
						"fullproductname": {
							"text": "Red Hat Software Collections for Red Hat Enterprise Linux Workstation (v. 6)",
							"productid": "6Workstation-RHSCL-3.3",
							"cpe": "cpe:/a:redhat:rhel_software_collections:3::el6"
						}
					},
					{
						"type": "Product Name",
						"name": "Red Hat Software Collections for Red Hat Enterprise Linux Server (v. 7)",
						"fullproductname": {
							"text": "Red Hat Software Collections for Red Hat Enterprise Linux Server (v. 7)",
							"productid": "7Server-Alt-RHSCL-3.3",
							"cpe": "cpe:/a:redhat:rhel_software_collections:3::el7"
						}
					},
					{
						"type": "Product Name",
						"name": "Red Hat Software Collections for Red Hat Enterprise Linux Server (v. 7)",
						"fullproductname": {
							"text": "Red Hat Software Collections for Red Hat Enterprise Linux Server (v. 7)",
							"productid": "7Server-RHSCL-3.3",
							"cpe": "cpe:/a:redhat:rhel_software_collections:3::el7"
						}
					},
					{
						"type": "Product Name",
						"name": "Red Hat Software Collections for Red Hat Enterprise Linux Server EUS (v. 7.5)",
						"fullproductname": {
							"text": "Red Hat Software Collections for Red Hat Enterprise Linux Server EUS (v. 7.5)",
							"productid": "7Server-RHSCL-3.3-7.5.Z",
							"cpe": "cpe:/a:redhat:rhel_software_collections:3::el7"
						}
					},
					{
						"type": "Product Name",
						"name": "Red Hat Software Collections for Red Hat Enterprise Linux Server EUS (v. 7.6)",
						"fullproductname": {
							"text": "Red Hat Software Collections for Red Hat Enterprise Linux Server EUS (v. 7.6)",
							"productid": "7Server-RHSCL-3.3-7.6.Z",
							"cpe": "cpe:/a:redhat:rhel_software_collections:3::el7"
						}
					},
					{
						"type": "Product Name",
						"name": "Red Hat Software Collections for Red Hat Enterprise Linux Server EUS (v. 7.7)",
						"fullproductname": {
							"text": "Red Hat Software Collections for Red Hat Enterprise Linux Server EUS (v. 7.7)",
							"productid": "7Server-RHSCL-3.3-7.7.Z",
							"cpe": "cpe:/a:redhat:rhel_software_collections:3::el7"
						}
					},
					{
						"type": "Product Name",
						"name": "Red Hat Software Collections for Red Hat Enterprise Linux Workstation (v. 7)",
						"fullproductname": {
							"text": "Red Hat Software Collections for Red Hat Enterprise Linux Workstation (v. 7)",
							"productid": "7Workstation-RHSCL-3.3",
							"cpe": "cpe:/a:redhat:rhel_software_collections:3::el7"
						}
					}
				],
				"fullproductname": {}
			},
			{
				"type": "Product Version",
				"name": "rh-python36-python-0:3.6.9-2.el6",
				"fullproductname": {
					"text": "rh-python36-python-3.6.9-2.el6.src.rpm",
					"productid": "rh-python36-python-0:3.6.9-2.el6",
					"cpe": "cpe:/a:redhat:rhel_software_collections:3::el6"
				}
			},
			{
				"type": "Product Version",
				"name": "rh-python36-python-0:3.6.9-2.el7",
				"fullproductname": {
					"text": "rh-python36-python-3.6.9-2.el7.src.rpm",
					"productid": "rh-python36-python-0:3.6.9-2.el7",
					"cpe": "cpe:/a:redhat:rhel_software_collections:3::el7"
				}
			}
		],
		"relationship": [
			{
				"productreference": "rh-python36-python-0:3.6.9-2.el6",
				"relationtype": "Default Component Of",
				"relatestoproductreference": "6Server-RHSCL-3.3",
				"fullproductname": {
					"text": "rh-python36-python-0:3.6.9-2.el6 as a component of Red Hat Software Collections for Red Hat Enterprise Linux Server (v. 6)",
					"productid": "6Server-RHSCL-3.3:rh-python36-python-0:3.6.9-2.el6",
					"cpe": "cpe:/a:redhat:rhel_software_collections:3::el6"
				}
			},
			{
				"productreference": "rh-python36-python-0:3.6.9-2.el6",
				"relationtype": "Default Component Of",
				"relatestoproductreference": "6Workstation-RHSCL-3.3",
				"fullproductname": {
					"text": "rh-python36-python-0:3.6.9-2.el6 as a component of Red Hat Software Collections for Red Hat Enterprise Linux Workstation (v. 6)",
					"productid": "6Workstation-RHSCL-3.3:rh-python36-python-0:3.6.9-2.el6",
					"cpe": "cpe:/a:redhat:rhel_software_collections:3::el6"
				}
			},
			{
				"productreference": "rh-python36-python-0:3.6.9-2.el7",
				"relationtype": "Default Component Of",
				"relatestoproductreference": "7Server-Alt-RHSCL-3.3",
				"fullproductname": {
					"text": "rh-python36-python-0:3.6.9-2.el7 as a component of Red Hat Software Collections for Red Hat Enterprise Linux Server (v. 7)",
					"productid": "7Server-Alt-RHSCL-3.3:rh-python36-python-0:3.6.9-2.el7",
					"cpe": "cpe:/a:redhat:rhel_software_collections:3::el7"
				}
			},
			{
				"productreference": "rh-python36-python-0:3.6.9-2.el7",
				"relationtype": "Default Component Of",
				"relatestoproductreference": "7Server-RHSCL-3.3",
				"fullproductname": {
					"text": "rh-python36-python-0:3.6.9-2.el7 as a component of Red Hat Software Collections for Red Hat Enterprise Linux Server (v. 7)",
					"productid": "7Server-RHSCL-3.3:rh-python36-python-0:3.6.9-2.el7",
					"cpe": "cpe:/a:redhat:rhel_software_collections:3::el7"
				}
			},
			{
				"productreference": "rh-python36-python-0:3.6.9-2.el7",
				"relationtype": "Default Component Of",
				"relatestoproductreference": "7Server-RHSCL-3.3-7.5.Z",
				"fullproductname": {
					"text": "rh-python36-python-0:3.6.9-2.el7 as a component of Red Hat Software Collections for Red Hat Enterprise Linux Server EUS (v. 7.5)",
					"productid": "7Server-RHSCL-3.3-7.5.Z:rh-python36-python-0:3.6.9-2.el7",
					"cpe": "cpe:/a:redhat:rhel_software_collections:3::el7"
				}
			},
			{
				"productreference": "rh-python36-python-0:3.6.9-2.el7",
				"relationtype": "Default Component Of",
				"relatestoproductreference": "7Server-RHSCL-3.3-7.6.Z",
				"fullproductname": {
					"text": "rh-python36-python-0:3.6.9-2.el7 as a component of Red Hat Software Collections for Red Hat Enterprise Linux Server EUS (v. 7.6)",
					"productid": "7Server-RHSCL-3.3-7.6.Z:rh-python36-python-0:3.6.9-2.el7",
					"cpe": "cpe:/a:redhat:rhel_software_collections:3::el7"
				}
			},
			{
				"productreference": "rh-python36-python-0:3.6.9-2.el7",
				"relationtype": "Default Component Of",
				"relatestoproductreference": "7Server-RHSCL-3.3-7.7.Z",
				"fullproductname": {
					"text": "rh-python36-python-0:3.6.9-2.el7 as a component of Red Hat Software Collections for Red Hat Enterprise Linux Server EUS (v. 7.7)",
					"productid": "7Server-RHSCL-3.3-7.7.Z:rh-python36-python-0:3.6.9-2.el7",
					"cpe": "cpe:/a:redhat:rhel_software_collections:3::el7"
				}
			},
			{
				"productreference": "rh-python36-python-0:3.6.9-2.el7",
				"relationtype": "Default Component Of",
				"relatestoproductreference": "7Workstation-RHSCL-3.3",
				"fullproductname": {
					"text": "rh-python36-python-0:3.6.9-2.el7 as a component of Red Hat Software Collections for Red Hat Enterprise Linux Workstation (v. 7)",
					"productid": "7Workstation-RHSCL-3.3:rh-python36-python-0:3.6.9-2.el7",
					"cpe": "cpe:/a:redhat:rhel_software_collections:3::el7"
				}
			}
		]
	}
}
